### PR TITLE
Allows any tenant to be at the root

### LINKED ISSUE #13952: AutoSetup issue with "/" RequestUrlPrefix
AutoSetup throws a validation error when trying to create a tenant with "/" RequestUrlPreix.

The ultimate intent is to create a default tenant at /default and the auth server tenant at /.

cc @jtkech

### PR DESCRIPTION
**Pull request by jtkech**
Fixes #13952 

Meaning to AutoSetup the Default tenant but with a non empty `RequestUrlPrefix`. And then also allows to Autosetup a non Default tenant being at the root, meaning having a null/empty `RequestUrlPrefix`.

Todo: Also allow the same from the admin UI or API controller, as it was before.

Okay done by updating the `TenantValidator` but we still check if 2 tenants (including the Default) have both the same `RequestUrlPrefix` and a shared `RequestUrlHost`.

### CODE REVIEW

**Comment by jtkech** [comment]
Has been tried by @devlife, see https://github.com/OrchardCMS/OrchardCore/issues/13952#issuecomment-1624559289

@Skrypt I let you review it when you will have time, and approve if you think that's okay.

**Review by Skrypt — APPROVED** [review]

**Comment by Skrypt** [comment]
Don't merge it yet. I want to have some time to test it. Need to see if we can automate some tests with this.

File: src/OrchardCore.Modules/OrchardCore.AutoSetup/AutoSetupMiddleware.cs
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.AutoSetup/AutoSetupMiddleware.cs:253]: Just before there is a CreateTenantSettings() which is called for each tenant to pre-create them with the Autosetup options, unless for the Default tenant which is still created by the shellHost initialization on the first request.

**Comment by jtkech** [comment]
Okay, let me know ;)

**Pull request merged**
